platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md

```diff
@@ -0,0 +1,202 @@
+branch in the hashgraph on restart by building a new self-event on the same self-event parent it had previously used.
+In other words, _a node must only tell the world of a self-event **after** it has recorded this information!_
+
+This proposal ensures self-events are persisted to disk before they are gossiped. It simplifies the relationship
+between the consensus engine and the PCES writer, removing the need for the PCES sequencer, the `RoundDurabilityBuffer`,
+and the responsibility of the consensus engine to tell the PCES writer when to flush, while introducing negligible
+additional latency.
+
+| Metadata           | Entities                            | 
+|--------------------|-------------------------------------|
+| Designers          | Cody, Richard, Jasper               |
+| Functional Impacts | Platform                            |
+| Related Proposals  | N/A                                 |
+| HIPS               | N/A                                 |
+
+---
+
+## Purpose and Context
+
+Nodes in the network create new "self-events" on a regular basis. These self-events carry transactions that were
+submitted to the node, and advance the hashgraph. When a node creates a self-event, it must build upon the previous
+self-event and some "other parent" event (see the hashgraph algorithm for more details). If a node builds two
+self-events on the same self-event parent or if it creates an event with no self parent and there is a non-ancient self
+event that could have been a self parent, that is considered to be a branch in the hashgraph. Branching in the
+hashgraph is considered an attack on consensus. It can be detected deterministically and the offending node punished
+accordingly, protecting the integrity of the hashgraph. Therefore, no honest node should ever branch.
+
+Today even honest nodes can accidentally branch (in practice the probability low but not unheard of). This happens
+when a node gossips a self-event and is shutdown before the event has been persisted to disk in the
+Pre-Consensus Event Stream (PCES). On restart, the PCES is replayed to populate the system with missing events,
+and other events that occurred since the node went down are retrieved via gossip. But there is a time between when
+the node starts up and sends a new self-event and when it receives its old self-event from other nodes. During this
+time, the node will build a new self-event on the same parent as the previous self-event (or with no self parent),
+causing a branch.
+
+This happens today most frequently on update boundaries when **every node** deterministically shutdown and then
+start back up. In theory it can also happen when individual nodes are restarted.
+
+### Requirements
+
+1. There must not be any noticeable impact in time-to-consensus.
+    a) if there is an observed impact, all product stakeholders must agree that the impact is negligible
```

Comment:
Why a) if there is no b). Next to that the formatting is broken: you need to add a newline to have it formatted correctly in MD.

Reply:
collapsed the sub bullets